### PR TITLE
Passage au level up (juste passage en mode opti)

### PR DESCRIPTION
**Pull request by Arno500**

### CODE REVIEW

**Comment by alexandre025** [comment]
Top, je vais pouvoir te faire une vraie review 👌 Ensuite tu pourras "squash and merge" depuis Github (= fusionner tout les commits en un et merge sur master)

File: backend/level1/main.rb [hunk 1]
```diff
@@ -1,25 +1,30 @@
 require 'date'
 require 'json'
+require './models/car.rb'
```
Comment (alexandre025) [inline backend/level1/main.rb]: You can use `require_relative`

File: backend/level1/models/rental.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+class Rental
+  attr_reader :id, :car, :days, :distance
+  def initialize(id, car, start_date, end_date, distance)
+    @id = id
+    @car = car
+    start_date = start_date
+    end_date = end_date
+    @distance = distance.to_i
+
+    @days = (Date.parse(end_date) - Date.parse(start_date) + 1).to_i
```
Comment (alexandre025) [inline backend/level1/models/rental.rb]: You can extract this into a public method `days`. You can refactor this using enumerables:
`(Date.parse(start_date)..Date.parse(end_date)).count` (`..` enumerate all values between to values in an array, it also work for integers)

File: backend/level1/models/rental.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def time_price
+    @days * @car.price_per_day
+  end
+
+  # Erreur ici
```
Comment (alexandre025) [inline backend/level1/models/rental.rb]: I won't spend too much time here as you probably fix it in the following PR

File: backend/level1/main.rb [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
+  cars.push(Car.new(car['id'], car['price_per_day'], car['price_per_km']))
+end
 
+output = Hash[rentals: []]
 
-data["rentals"].each do |elm|
-    wantedCar = data["cars"][data["cars"].index{ |x| x["id"] == elm["car_id"]}]
-    dateDifference = Date.parse(elm["end_date"]) - Date.parse(elm["start_date"]) + 1
+data['rentals'].each do |rental|
+  wanted_car = cars.find { |car| car.id == rental['car_id'] }
```
Comment (alexandre025) [inline backend/level1/main.rb]: Nice uses of blocks 👍

File: backend/level1/models/rental.rb (hunk 2 above)
Comment (Arno500) [reply backend/level1/models/rental.rb]: What is the goal if I just need the number of days between, without realizing any operation on each number?

Comment (Arno500) [reply backend/level1/models/rental.rb]: Isn't it slower than a 'simple' math operation optimized by the Date module?

File: backend/level1/models/rental.rb (hunk 3 above)
Comment (Arno500) [reply backend/level1/models/rental.rb]: In fact there was no error, I removed the comment XD

File: backend/level1/models/rental.rb (hunk 2 above)
Comment (alexandre025) [reply backend/level1/models/rental.rb]: using `-` on dates is a bit more complex than a substraction on integers. Here, using enumerable is clearer be cause you don't have to add a `+1`

Comment (Arno500) [reply backend/level1/models/rental.rb]: Oki ! Fixing...

**Pull request merged**